src/include/migraphx/op/reshape.hpp

```diff
@@ -215,32 +231,51 @@ struct reshape
 
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this, true}.has(1);
+        check_shapes{inputs, *this, true}.has(1, 2);
 
         auto n_neg_dims = std::count(dims.begin(), dims.end(), -1);
         if(n_neg_dims > 1)
             MIGRAPHX_THROW("reshape: Dimensions for reshape can only have one -1 dim");
 
         auto s0 = inputs.front();
-        if(s0.dynamic())
+        if(inputs.size() == 1)
         {
-            return dyn_compute_shape(s0);
+            if(s0.dynamic())
+            {
+                return dyn_compute_shape(s0);
+            }
+            else
+            {
+                return static_compute_shape(inputs, n_neg_dims);
+            }
         }
         else
         {
-            return static_compute_shape(inputs, n_neg_dims);
+            return inputs.back();
         }
     }
 
     argument compute(const dyn_output& dyn_out, std::vector<argument> args) const
     {
         assert(dyn_out.computed_shape.standard());
-        argument result{dyn_out.computed_shape};
+        if(args.size() == 1)
+        {
+            argument result{dyn_out.computed_shape};
 
-        visit_all(result, args[0])([&](auto output, auto input) {
-            std::copy(input.begin(), input.end(), output.begin());
-        });
-        return result;
+            visit_all(result, args[0])([&](auto output, auto input) {
+                std::copy(input.begin(), input.end(), output.begin());
+            });
+            return result;
+        }
+        else
+        {
+            // 2 arg
+            assert(args[0].get_shape().elements() == args[1].get_shape().elements());
```

Comment:
It should throw instead of assert. If elements do not match then it is a runtime error.

Reply:
can you add verify_onnx where it would throw?